packages/core/utils/src/package-manager.ts

```diff
@@ -0,0 +1,37 @@
+import execa from 'execa';
+import preferredPM from 'preferred-pm';
+
+import type { Options as ProcessOptions } from 'execa';
+
+const SUPPORTED_PACKAGE_MANAGERS = ['npm', 'yarn'];
+const DEFAULT_PACKAGE_MANAGER = 'npm' as const;
+
+type SupportedPackageManagerName = 'npm' | 'yarn';
+
+export const getPreferred = async (pkgPath: string): Promise<SupportedPackageManagerName> => {
+  const pm = await preferredPM(pkgPath);
+
+  const hasPackageManager = pm !== undefined;
+  if (!hasPackageManager) {
+    throw new Error(`Couldn't find a package manager in your project.`);
+  }
+
+  const isPackageManagerSupported = SUPPORTED_PACKAGE_MANAGERS.includes(pm.name);
+  if (!isPackageManagerSupported) {
+    process.emitWarning(
+      `We detected your package manager (${pm.name} v${pm.version}), but it's not officially supported by Strapi yet. Defaulting to npm instead.`
+    );
+
+    return DEFAULT_PACKAGE_MANAGER;
```

Comment:
Do you think this is the right approach? I wonder if it's more about saying "you're using something we don't official support, if it breaks it's not our problem" – I feel like this is quite defensive DX but in reality, we should really be more flexible and just warn them we don't guarantee it.

For example, we do this on the FE with react-router-dom / styled-components / react / react-dom etc. We recommend you use the versions we specify, but if you don't and say you use V6 of styled-components now, we won't stop you but we warn you. 🤔

Reply:
We have had an interesting discussion about this with @innerdvations.

Our approach here would be to be more on the defensive side for the MVP so that we have time to analyze usage and main issues during the early stage of the release, and then iterate on this to allow more flexibility.

From a personal point of view, I don't mind both approaches as everything should just "work" as long as the detected package manager has an `install` command. We could just go with just a warning & proceed anyway

Reply:
My main concern is issues being opened about unsupported features & time spent on this by core contributors instead of actual bugs.

Reply:
> My main concern is issues being opened about unsupported features & time spent on this by core contributors instead of actual bugs.

It's valid concern, but lets not lie, we were getting issues continuously about pnpm not working even though we said countless times it wasn't supported 😂  but i do understand what you mean.

Reply:
> I would suggest merging the current approach for now and further discussing this, then potentially changing the behavior before the GA.

we are two suggesting it now :)